### PR TITLE
jint jsvalue works, no need to use json.parse as workaround

### PR DESCRIPTION
**Pull request by vicancy**
@vwxyzh @herohua @superyyrrzz @chenkennt

### CODE REVIEW

File: src/Microsoft.DocAsCode.Utility/ConvertToObjectHelper.cs [hunk 1]
```diff
@@ -43,5 +46,22 @@ public static object ConvertJObjectToObject(object raw)
...
+            using (MemoryStream ms = new MemoryStream())
+            {
+                using (StreamWriter sw = new StreamWriter(ms))
+                {
+                    JsonUtility.Serialize(sw, raw);
+                    sw.Flush();
+                    ms.Seek(0, SeekOrigin.Begin);
+                    using (StreamReader sr = new StreamReader(ms))
+                    {
+                        return ConvertJObjectToObject(JsonUtility.Deserialize<object>(sr));
```
Comment (chenkennt) [inline src/Microsoft.DocAsCode.Utility/ConvertToObjectHelper.cs]: directly convert to jsvalue?

File: Documentation/template/index.html.js [hunk 2]
```diff
@@ -1,18 +1,15 @@
 function transform(model, _attrs){
```
Comment (chenkennt) [inline Documentation/template/index.html.js]: _attrs -> attrs

**Comment by superyyrrzz** [comment]
Should we add updated template zip to this commit?

File: Documentation/template/index.html.js (hunk 2 above)
Comment (vicancy) [inline Documentation/template/index.html.js]: I am thinking using underscore prefix to stand for a `docfx` generated property..

File: src/Microsoft.DocAsCode.Utility/ConvertToObjectHelper.cs (hunk 1 above)
Comment (vicancy) [inline src/Microsoft.DocAsCode.Utility/ConvertToObjectHelper.cs]: add a JsonConverter

File: src/Microsoft.DocAsCode.EntityModel/TemplateManagers/Template.cs [hunk 3]
```diff
@@ -63,41 +57,30 @@ public string Transform(string modelPath, object attrs)
...
             return _renderer.Render(model);
         }
 
-        private string GetRelativeResourceKey(string relativePath)
+        private object ProcessWithJint(object model, object attrs)
         {
-            // Make sure resource keys are combined using '/'
-            return Path.GetDirectoryName(this.Name).ToNormalizedPath().ForwardSlashCombine(relativePath);
+            var argument1 = JintProcessorHelper.ConvertStrongTypeToJsValue(model);
+            var argument2 = JintProcessorHelper.ConvertStrongTypeToJsValue(attrs);
```
Comment (chenkennt) [inline src/Microsoft.DocAsCode.EntityModel/TemplateManagers/Template.cs]: In some cases it's already JObject? no need to serialize again

File: src/Microsoft.DocAsCode.EntityModel/TemplateManagers/JintProcessorHelper.cs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            }
+
+            public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+            {
+                if (reader.TokenType == JsonToken.StartArray)
+                {
+                    var jArray = JArray.Load(reader);
+                    return ConvertJObjectToJsValue(jArray);
+                }
+                else
```
Comment (chenkennt) [inline src/Microsoft.DocAsCode.EntityModel/TemplateManagers/JintProcessorHelper.cs]: Support primitive type?

File: src/docfx.website.themes/docs.html/ManagedReference.mta.json.js [hunk 5]
```diff
@@ -1,7 +1,4 @@
-function transform(model, _attrs){
-  var entity = JSON.parse(model);
-  var attrs = JSON.parse(_attrs);
-
+function transform(entity, attrs){
```
Comment (chenkennt) [inline src/docfx.website.themes/docs.html/ManagedReference.mta.json.js]: not consistent with your naming convention

**Comment by chenkennt** [comment]
:+1:

**Pull request closed without merging**